stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/attachments/content/MediaAttachmentContent.kt

```diff
@@ -0,0 +1,485 @@
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.heightIn
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.material.Text
+import androidx.compose.material.ripple.rememberRipple
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import coil.compose.AsyncImagePainter
+import coil.request.ImageRequest
+import com.getstream.sdk.chat.utils.extensions.imagePreviewUrl
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.models.Attachment
+import io.getstream.chat.android.client.models.ConnectionState
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.compose.R
+import io.getstream.chat.android.compose.state.mediagallerypreview.MediaGalleryPreviewResult
+import io.getstream.chat.android.compose.state.messages.attachments.AttachmentState
+import io.getstream.chat.android.compose.ui.attachments.preview.MediaGalleryPreviewContract
+import io.getstream.chat.android.compose.ui.components.MediaPreviewPlaceHolder
+import io.getstream.chat.android.compose.ui.theme.ChatTheme
+import io.getstream.chat.android.compose.ui.util.rememberStreamImagePainter
+import io.getstream.chat.android.uiutils.constant.AttachmentType
+import io.getstream.chat.android.uiutils.extension.hasLink
+
+/**
+ * Displays a preview of single or multiple video or attachments.
+ *
+ * @param attachmentState The state of the attachment, holding the root modifier, the message
+ * and the onLongItemClick handler.
+ * @param modifier The modifier used for styling.
+ * @param maximumNumberOfPreviewedItems The maximum number of thumbnails that can be displayed
+ * in a group when previewing Media attachments in the message list.
+ * @param playButton Represents the play button that is overlaid above video attachment
+ * previews.
+ */
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+public fun MediaAttachmentContent(
+    attachmentState: AttachmentState,
+    modifier: Modifier = Modifier,
+    maximumNumberOfPreviewedItems: Int = 4,
+    playButton: @Composable () -> Unit = { PlayButton() },
```

Comment:
This is a bit weird - I'd maybe rename this to playbackContent or something like that, maybe actionContent 🤷 

Because playButton is too specific and people might want to replace this with another action other than play (not sure what specifically it might be, but you get the idea).

E.g. if it's images, they might add the "Preview Icon" that just zooms in on the image without opening the gallery picker or something

Reply:
That's a good point.

For starters, I'm going to do this

- Rename to `overlayContent`
- Change signature to `@Composable (attachmentType: Unit) -> Unit`

The second point is because currently **we** only show it over videos, but using this approach we can let the users show it over whatever they want.